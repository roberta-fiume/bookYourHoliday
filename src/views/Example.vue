<template>
    <div>
        <v-container>
          <div class="hello2">Hello Div 2</div>
          <input class="inputField2" placeholder="abc2">
        </v-container>
        <div class="hello">Hello Div</div>
        <input class="inputField" placeholder="abc">
    </div>
</template>

<script>
export default {
    data() {
        return {
            t1: null,
            t2: null,
            errorMessage: false,
        }
    }
}
</script>