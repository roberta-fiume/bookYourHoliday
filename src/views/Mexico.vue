<template>
<div>
   <h1> MEXICO</h1>
    <v-carousel>
        <v-carousel-item
        v-for="(item,i) in items"
        :key="i"
        :src="item.src"
        reverse-transition="fade-transition"
        transition="fade-transition"
        ></v-carousel-item>
    </v-carousel> 

    <div class="booking-bedroom-size">
        <span>Bedroom: </span>
        <v-container fluid>
            <v-radio-group v-model="radios" :mandatory="false">
            <v-radio class="radioButtonBedroomSize" label="Single" value="Single"></v-radio>
            <v-radio class="radioButtonBedroomSize" label="Double" value="Double"></v-radio>
            </v-radio-group>
            <p class="bedroom-size-text">You chose: {{ radios || 'null' }}</p>
          <p> Please enter your name:</p> <input v-model="firstName" placeholder="Enter your name" class="inputField">
          <p> Please enter your last name:</p> <input v-model="lastName" placeholder="Enter your last name" class="inputField">
        </v-container>
    </div> 

    <div class="booking-num-people">
        <span>Number of people: {{people}}</span>
        <v-btn class="butt-increment" @click="people++"> + </v-btn>
        <v-btn class="butt-decrement" @click="people--"> - </v-btn>
    </div>

    <div class="booking-bedroom-type">
        <span>Bedroom type: </span>
        <v-container fluid>
            <v-radio-group v-model="bedroomType" :mandatory="false">
            <v-radio class="radioButtonBedroomType" label="Standard" value="Standard"></v-radio>
            <v-radio class="radioButtonBedroomType" label="Suite" value="Suite"></v-radio>
            <v-radio class="radioButtonBedroomType" label="Gold Suite" value="Gold Suite"></v-radio>
            </v-radio-group>
            <p class="bedroom-type-text"> You chose: {{ bedroomType || 'null' }}</p>
        </v-container>
    </div>

    <v-btn class="showPrice" @click="showPrice">Show price</v-btn>

    <p class="errorMessage">{{errorMessage}}</p>

    <div class="booking-price" v-if="priceBox" >
        The booking price is: <span> {{finalPrice}}</span>
        <h2>YOUR BOOKING INFORMATION: </h2>
        <p>Your NAME is: {{infoPerson.name}} </p>
        <p>Your LAST NAME is: {{infoPerson.lastName}} </p>
        <p> Number of people: {{infoPerson.numberPeople}} </p>
        <p> Bedroom Size: {{infoPerson.bedroomSize}} </p>
        <p> Bedroom Type: {{infoPerson.bedroomType}} </p>
        <p> Price is: {{finalPrice}} </p>
    </div> 

    <div>
      <span>Are you happy with the booking?</span>
          <v-radio-group v-model="confirmation" :mandatory="false">
            <div :class="{hideRadioButton:hideRadioButton}">
                 <v-radio id="radioYes" label="Yes" value="Yes"  @change="goToPayment"></v-radio>
            </div>
            <div :class="{hideRadioButton:hideRadioButton}">
                <v-radio id="radioNo" label="No"  value="No"  @change="showButtonEditInfo"></v-radio>
            </div>
         
          </v-radio-group>
        <v-btn v-if="happy" @click="showRouter"><router-link to="/mexico/payment" > Go to payment</router-link></v-btn>
        <router-view :infoPersonProp="infoPerson" :exampleProp="example" v-if="buttonNav"></router-view>
        <v-btn v-if="notHappy"> Edit Information </v-btn>
    </div>

</div>
</template>

<script>
  export default {
    data () {
      return {
        items: [
          {
            src: require('@/assets/img/mexico.jpeg'),
          },
          {
            src: require('@/assets/img/mexico2.jpg'),
          },
          {
            src: require('@/assets/img/mexico3.jpg'),
          },
          {
            src: require('@/assets/img/mexico4.jpg'),
          }
        ],

        radios: 'Single',
        firstName: null,
        lastName: null,
        people: 0,
        bedroomType: 'Standard',
        finalPrice: '',
        errorMessage: '',
        priceBox: false,
        prices: null,
        infoPerson: {},
        confirmation: '',
        happy: false,
        notHappy: false,
        buttonNav: false,
        example: "helloo",
        hideRadioButton: false
      }
    },
    
    created() {
        this.people = 0;
    },

   methods: {
       showPrice() {
           this.priceBox = !this.priceBox;
           console.log("THIS IS MY STATE",this.priceBox)
           this.createPrices();
           this.clear();
           if (this.priceBox === true) {
             this.errorMessage = null;
           }
       },

       createPrices() {
          let bookingInfo = [];
          bookingInfo.push(this.radios, this.people, this.bedroomType, this.firstName, this.lastName,);
          this.prices = bookingInfo;
          console.log("my arrayyy", this.prices);
          this.combinations();
          this.throwError();
          this.createObjPerson();
       },

       combinations() {
         if (this.prices[0] === "Single" && this.prices[1] == 1 && this.prices[2] === "Standard") {
             this.finalPrice = '£50';
             console.log("FINAL PRICEEE", this.finalPrice)    
         } else if (this.prices[0] === "Single" && this.prices[1] == 1 && this.prices[2] === "Suite") {
             this.finalPrice = "We're sorry, we don't have Suites for single bedrooms";
        } else if (this.prices[0] === "Single" && this.prices[1] == 2 && this.prices[2] === "Standard") {
             this.finalPrice = "Error! there cannot be two people in a Single bedroom. Please retry.";
        }  else if (this.prices[0] === "Single" && this.prices[1] == 1 && this.prices[2] === "Gold Suite") {
             this.finalPrice = "We're sorry, we don't provide Gold Suites for single bedrooms";
         } else if (this.prices[0] === "Double" && this.prices[1] == 1 && this.prices[2] === "Standard") {
             this.finalPrice = '£100';
         } else if (this.prices[0] === "Double" && this.prices[1] == 1 && this.prices[2] === "Standard") {
              this.finalPrice = '£100';
         } else if (this.prices[0] === "Double" && this.prices[1] == 1 && this.prices[2] === "Suite") {
               this.finalPrice = '£170';
         } else if (this.prices[0] === "Double" && this.prices[1] == 1 && this.prices[2] === "Gold Suite") {
              this.finalPrice = '£220';
         } else if (this.prices[0] === "Double" && this.prices[1] == 2 && this.prices[2] === "Standard") {
             this.finalPrice = '£200';
         } else if (this.prices[0] === "Double" && this.prices[1] == 2 && this.prices[2] === "Suite") {
              this.finalPrice = '£270';
         } else if (this.prices[0] === "Double" && this.prices[1] == 2 && this.prices[2] === "Gold Suite") {
               this.finalPrice = '£350';
         }
       },

       throwError() {
         if (this.prices[3] === null || this.prices[4] === null || this.prices[1] == 0) {
            this.errorMessage = "YOUR INFORMATION IS NOT CORRECT. PLEASE RETRY!";
            this.priceBox = !this.priceBox;
            this.clear();
         } else {
            this.errorMessage = null;
         }
       },

       createObjPerson() {
         this.infoPerson.name = this.prices[3];
         this.infoPerson.lastName = this.prices[4];
         this.infoPerson.bedroomSize = this.prices[0];
         this.infoPerson.numberPeople = this.prices[1];
         this.infoPerson.bedroomType = this.prices[2];
         console.log("THIS IS THE OBJECTTT", this.infoPerson)

         return this.infoPerson
       },

       clear() {
         this.firstName = null;
         this.lastName = null;
       },

       goToPayment() {
         this.happy = !this.happy;
         this.notHappy = false;
         this.hideRadioButton = true;
       },

       showButtonEditInfo() {
         this.notHappy = !this.notHappy;
         this.happy = false;
       },

       showRouter() {
         this.buttonNav = !this.buttonNav;
       }
   }


  }
</script>

<style>
  .inputField {
    border-bottom: 2px solid lightseagreen;
  }

  .hideRadioButton {
    display: none;
  }

  
</style>



