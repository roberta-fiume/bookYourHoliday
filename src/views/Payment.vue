<template>
    <div>PAYMENT</div>
</template>

<script>
export default {
  props:['infoPersonProp'],

    data() {
        return {

        }
    },

    created() {
        console.log("I AM passedd", this.infoPersonProp)
    }
}
</script>