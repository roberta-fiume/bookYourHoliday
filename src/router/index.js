import Vue from 'vue'
import VueRouter from 'vue-router'
import Register from '../views/Register.vue'
import Destinations from '../views/Destinations.vue'
import Mexico from '../views/Mexico.vue'
import Dubai from '../views/Dubai.vue'
import Hawaii from '../views/Hawaii.vue'
import Maldives from '../views/Maldives.vue'
import Sardinia from '../views/Sardinia.vue'
import Payment from '../views/Payment.vue'
import Example from '../views/Example.vue'

Vue.use(VueRouter)

const routes = [
  {
    path: '/register',
    name: 'register',
    component: Register
  },
  {
    path: '/login',
    name: 'login',
    component: () => import('../views/Login.vue')
  },

  {
    path: '/',
    name: 'destinations',
    component: Destinations
  },

  {
    path: '/mexico',
    name: 'mexico',
    component: Mexico,

    children: [
      { path: 'payment', component: Payment}
    ]
  },

  {
    path: '/dubai',
    name: 'dubai',
    component: Dubai
  },

  {
    path: '/hawaii',
    name: 'hawaii',
    component: Hawaii
  },

  {
    path: '/maldives',
    name: 'maldives',
    component: Maldives
  },

  {
    path: '/sardinia',
    name: 'sardinia',
    component: Sardinia
  },

  {
    path: '/payment',
    name: 'payment',
    component: Payment
  },

  {
    path: '/example',
    name: 'example',
    component: Example
  }
]

const router = new VueRouter({
  mode: 'history',
  base: process.env.BASE_URL,
  routes
})

export default router
