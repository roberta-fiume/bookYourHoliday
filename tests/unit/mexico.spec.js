import { mount, shallowMount, createLocalVue } from '@vue/test-utils'
import Vue from 'vue';
import Vuetify from 'vuetify'
import VueRouter from 'vue-router';
import Mexico from "@/views/Mexico.vue"
import Payment from "@/views/Payment.vue"
import router from "@/router/index.js"
import Example from "@/views/Example.vue"

Vue.use(Vuetify);

const localVue = createLocalVue();

describe('Mexico', () => {
  // let vuetify = new Vuetify();
  let vuetify;
  localVue.use(VueRouter);

  const routes = [
    { path: '/payment', name: 'payment' }
  ];

  const router = new VueRouter({ routes });

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  it('displays h1 element', () => {
    let wrapper = shallowMount(Mexico, {
      localVue,
      vuetify,
      router,
    });
    // expect(wrapper.html()).toMatchSnapshot()
   // wrapper.setData({ items: [{ id: 1, name: 'My Item' }] });
    expect(wrapper.find('h1').exists()).toBe(true);
   });

});

// describe('Mexico', () => {
//   it('cheks that radio button for bedroom SIZE is selected', () => {
//     const wrapper = shallowMount(Mexico);
//     const radio = wrapper.find('.radioButtonBedroomSize');
//     radio.element.selected = true;
//     radio.trigger('change');
//   })
// });

// describe('Mexico', () => {
//   it('checks that v-container exists', () => {
//     const wrapper = shallowMount(Mexico);
//     expect(wrapper.find('.v-container').exists()).toBe(true);
//   })
// });


// function getValueFromInput() {
//   const wrapper = shallowMount(Mexico);
//   const value = wrapper.find('.inputField').element.value;
//   console.log("THIS IS THE VALUE", value);

//   return value
// }

// describe('Mexico', () => {
//   it('cheks that button Show Price shows the error message when input is null', () => {
//     const wrapper = shallowMount(Mexico);
//     // wrapper.setData({firstName: "roberta"})
//     // {
//     //   data() {
//     //     return {
//     //       t1: null,
//     //       t2: null,
//     //       errorMessage: false,
//     //     }
//     //   },
//     //   template: `
//     //   <div>
//     //     <v-container>
//     //       <div class="hello2">Hello Div 2</div>
//     //       <input class="inputField2" placeholder="abc2">
//     //     </v-container>
//     //     <div class="hello">Hello Div</div>
//     //     <input class="inputField" placeholder="abc">
//     //   </div>`
//     // }
    
//     // const wrapperArray = wrapper.findAll('.foo');
//     // console.log("arrayyy",wrapperArray);
//     // const newDiv = wrapper.find('.hello');
//     // console.log("divvv", newDiv.text());
//     // const newDiv2 = wrapper.find('.hello2');
//     // console.log("divv 2", newDiv2.text());
//     let expectedFirstName = "roberta123";
//     const inputValue = wrapper.find('.inputField');
//     inputValue.setValue(expectedFirstName);

    
//     // console.log(wrapper.vm.$data.firstName)
//     // console.log(wrapper.vm.firstName)
//     // console.log("value1", inputValue.attributes().value);

//     // const inputValueAfter = wrapper.find('.inputField');

//     expect(wrapper.vm.firstName).toEqual(expectedFirstName);
//     // console.log("value1 AFTER!!!!!", inputValueAfter.);

//     // const inputValue2 = wrapper.find('.inputField2');
//     // console.log("value2", inputValue2.attributes().value);

//     // console.log(wrapper.find('.first-name').text());

//     // expect(wrapper.vm.t1).toEqual(null);
//     // expect(wrapper.vm.t2).toEqual(null);
//     // wrapperArray.setValue(null);
//     // wrapper.find('.showPrice').trigger('click');
//     // expect(wrapper.find('.errorMessage').text()).toEqual('YOUR INFORMATION IS NOT CORRECT. PLEASE RETRY!')
//     // expect(wrapper.vm.t2).toEqual('foo');
    
//   })
// })

fdescribe('Testing input fields', () => {
  it('should use default value for firstName and show error when showPrice is clicked without using the input field', () => {
    const wrapper = shallowMount(Mexico);

    wrapper.find('.showPrice').trigger('click');

    expect(wrapper.vm.firstName).toBe(null);
    expect(wrapper.find('.errorMessage').text()).toEqual('YOUR INFORMATION IS NOT CORRECT. PLEASE RETRY!');    
  })

  it('should have empty string as input value and show error message when input is set as null', () => {
    const wrapper = shallowMount(Mexico);
    const inputValue = wrapper.find('.inputField');
    inputValue.setValue(null);

    wrapper.find('.showPrice').trigger('click');

    expect(wrapper.vm.firstName).toBe("");
    expect(wrapper.find('.errorMessage').text()).toEqual('YOUR INFORMATION IS NOT CORRECT. PLEASE RETRY!');    
  })

  fit('should NOT show error message when input is set as a valid value', () => {
    const wrapper = shallowMount(Mexico);
    let expectedFirstName = "roberta123";
    const inputValue = wrapper.find('.inputField');
    inputValue.setValue(expectedFirstName);

    wrapper.find('.showPrice').trigger('click');

    expect(wrapper.vm.firstName).toBe(expectedFirstName);
    expect(wrapper.find('.errorMessage').exists()).toBe(false);
  })
})

// fdescribe('Mexico', () => {
//   it('cheks that button Show Price shows the error message when input is null', () => {
//     const wrapper = shallowMount(Mexico);
//     let arrayInput = wrapper.findAll('div');
//     console.log("ARRAY", arrayInput);
    
//     arrayInput.forEach(input => input.setValue('null'));

//     wrapper.find('.showPrice').trigger('click');
//     expect(wrapper.find('.errorMessage').text()).toEqual('YOUR INFORMATION IS NOT CORRECT. PLEASE RETRY!')
//   })
// })

  // describe('Mexico', () => {
  //   it('checks that input fields change when user types', () => {
  //     const wrapper = shallowMount(Mexico);
  //     const value = wrapper.find('.inputField').element.value;
  //     console.log("THIS IS THE VALUE", value);
  //   })
  // })Â¬


  // describe('Mexico', () => {
  //   it('checks that input fields change when user types', () => {
  //     addInputValue('aValue');
  //   })
  // });

  // function addInputValue(value) {
  //   const wrapper = shallowMount(Mexico);
  //   let valueFromInput = wrapper.find('.inputField');
  //   valueFromInput.element.value = value;
  //   valueFromInput.trigger('input');

  //   wrapper.find('.showPrice').trigger('click');
  // }

  describe('Mexico', () => {
    it('increments counter value when button is clicked', () => {
      let vuetify = new Vuetify();
      const wrapper = shallowMount(Mexico, {
        vuetify
      });
      expect(wrapper.find('.booking-num-people > span').text()).toEqual('Number of people: 0');
      const incrementPeople = jest.fn();
      wrapper.setMethods({incrementPeople});
      const button = wrapper.find('.butt-increment');
      button.trigger('click');
      wrapper.vm.$emit('foo')
      expect(incrementPeople).toHaveBeenCalledTimes(1);
    })
  });

  describe('Mexico', () => {
    it('cheks that radio button for bedroom TYPE is selected', () => {
      const wrapper = shallowMount(Mexico);
      const radio = wrapper.find('.radioButtonBedroomType');
      radio.element.selected = true;
      radio.trigger('change');
    })
  });

  describe('Mexico', () => {
    it('checks that button Show Price shows the box with booking info', () => {
      const wrapper = shallowMount(Mexico);
      wrapper.find('.showPrice').trigger('click');
      expect(wrapper.find('.booking-price').exists()).toBe(true);
    })
  });

  // describe('Mexico', () => {
  //   it('cheks that button Show Price shows the error message when input is null', () => {
  //     const wrapper = shallowMount(Mexico);
  //     addInputValue(null);
  //     // expect(wrapper.find('.showPrice').trigger('click'));
  //     expect(wrapper.find('.errorMessage').exists()).toBe(true);
  //   })
  // });

  describe('Payment', () => {
    it('cheks that prop exists in Payment Component', () => {
      let expectedProp = {
        name: 'julio',
        lastName: 'fiume',
        bedroomSize: 'Double',
        numberPeople: '2',
        bedroomType: 'Suite'
      }; 
      
      const wrapper = shallowMount(Payment, {
        propsData: {
          infoPersonProp: expectedProp
        }
      });

      console.log("THIS IS THE NAMEEE", wrapper);
      expect(wrapper.props().infoPersonProp).toEqual(expectedProp);
      expect(wrapper.find('.name').text()).toEqual('julio');
      // expect(wrapper.props('infoPersonProp').name).toBe('roberta')
    })
  });

// describe('Mexico', () => {
//   it('tests that Create Object Person function is called'), () => {
//     const mockFunct = jest.fn();
//     const createObjectFn = createObjPerson(mockFunct);
//     expect(mockFunct).ToHaveBeenCalled();
//   }
// })
