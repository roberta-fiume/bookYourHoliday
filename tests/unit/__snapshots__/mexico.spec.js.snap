// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mexico displays h1 element 1`] = `
<div>
  <h1> MEXICO</h1>
  <v-carousel></v-carousel>
  <div class="booking-bedroom-size"><span>Bedroom: </span>
    <v-container fluid="">
      <v-radio-group>
        <v-radio label="Single" value="Single" class="radioButtonBedroomSize"></v-radio>
        <v-radio label="Double" value="Double" class="radioButtonBedroomSize"></v-radio>
      </v-radio-group>
      <p class="bedroom-size-text">You chose: Single</p>
      <p> Please enter your name:</p> <input placeholder="Enter your name" class="inputField">
      <p> Please enter your last name:</p> <input placeholder="Enter your last name" class="inputField">
    </v-container>
  </div>
  <div class="booking-num-people"><span>Number of people: 0</span>
    <v-btn class="butt-increment"> + </v-btn>
    <v-btn class="butt-decrement"> - </v-btn>
  </div>
  <div class="booking-bedroom-type"><span>Bedroom type: </span>
    <v-container fluid="">
      <v-radio-group>
        <v-radio label="Standard" value="Standard" class="radioButtonBedroomType"></v-radio>
        <v-radio label="Suite" value="Suite" class="radioButtonBedroomType"></v-radio>
        <v-radio label="Gold Suite" value="Gold Suite" class="radioButtonBedroomType"></v-radio>
      </v-radio-group>
      <p class="bedroom-type-text"> You chose: Standard</p>
    </v-container>
  </div>
  <v-btn class="showPrice">Show price</v-btn>
  <p class="errorMessage"></p>
  <!---->
  <div><span>Are you happy with the booking?</span>
    <v-radio-group>
      <div class="">
        <v-radio id="radioYes" label="Yes" value="Yes"></v-radio>
      </div>
      <div class="">
        <v-radio id="radioNo" label="No" value="No"></v-radio>
      </div>
    </v-radio-group>
    <!---->
    <!---->
    <!---->
  </div>
</div>
`;
